src/sentry/ingest/ingest_consumer.py

```diff
@@ -84,25 +88,46 @@ def flush_batch(self, batch):
         with metrics.timer("ingest_consumer.fetch_projects"):
             projects = {p.id: p for p in Project.objects.get_many_from_cache(projects_to_fetch)}
 
-        # attachment_chunk messages need to be processed before attachment/event messages.
-        with metrics.timer("ingest_consumer.process_attachment_chunk_batch"):
-            for _ in self.pool.imap_unordered(
-                lambda msg: process_attachment_chunk(msg, projects=projects),
-                attachment_chunks,
-                chunksize=100,
-            ):
-                pass
-
-        with metrics.timer("ingest_consumer.process_other_messages_batch"):
-            for _ in self.pool.imap_unordered(
-                lambda args: args[0](args[1], projects=projects), other_messages, chunksize=100
-            ):
-                pass
+        if attachment_chunks:
+            # attachment_chunk messages need to be processed before attachment/event messages.
+            with metrics.timer("ingest_consumer.process_attachment_chunk_batch"):
+                for _ in self.pool.imap_unordered(
+                    lambda msg: process_attachment_chunk(msg, projects=projects),
+                    attachment_chunks,
+                    chunksize=100,
+                ):
+                    pass
+
+        if other_messages:
+            with metrics.timer("ingest_consumer.process_other_messages_batch"):
+                for _ in self.pool.imap_unordered(
+                    lambda args: args[0](args[1], projects=projects), other_messages, chunksize=100
+                ):
+                    pass
+
+        if transactions:
+            process_transactions_batch(transactions, projects)
 
     def shutdown(self):
         pass
 
 
+@metrics.wraps("ingest_consumer.process_transactions_batch")
+def process_transactions_batch(messages, projects):
+    events = []
+    for message in messages:
```

Comment:
Do you want to time this loop or parts of it?

Reply:
it's just json decode so i am timing that now